### PR TITLE
[Enhancement] (nereids)implement showEncryptKeysCommand in nereids

### LINKED ISSUE #42774: [Enhancement] (nereids)implement showEncryptKeysCommand in nereids
showEncryptKeys command is supported in legacy planner but not in nereids(new doris fe planner). you need implement the following showEncryptKeys command in nereids:
unsupportedshowEncryptKeysStatement
``` SHOW ENCRYPTKEYS ((FROM | IN) database=multipartIdentifier)? wildWhere?       ```

### PR DESCRIPTION
**Pull request by msridhar78**
### What problem does this PR solve?

Issue Number: close #42774 

Related PR: #xxx

Problem Summary:

### Release note

None

### Check List (For Author)

- Test 
    - [x] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by msridhar78** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowEncryptKeysCommand.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+ */
+public class ShowEncryptKeysCommand extends ShowCommand {
+    private static final ShowResultSetMetaData META_DATA =
+            ShowResultSetMetaData.builder()
+                    .addColumn(new Column("EncryptKey Name", ScalarType.createVarchar(20)))
+                    .addColumn(new Column("EncryptKey String", ScalarType.createVarchar(1024)))
+                    .build();
+
+    private String dbName;
+    private String wild;
```
Comment (Vallishp) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowEncryptKeysCommand.java:59]: can we make wild as final?

Comment (msridhar78) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowEncryptKeysCommand.java]: done - please review

**Comment by msridhar78** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowEncryptKeysCommand.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+                }
+                resultRowSet.add(resultRow);
+                encryptKeyNameSet.add(resultRow.get(0));
+            }
+        }
+
+        return new ShowResultSet(META_DATA, resultRowSet);
+    }
+
+    public boolean like(String str) {
```
Comment (starocean999) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowEncryptKeysCommand.java:120]:
```suggestion
    private boolean like(String str) {
```
better to change it as private method, you can do it in your next other pr :smile:

**Review by starocean999 — DISMISSED** [review]

**Review by Vallishp — APPROVED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowEncryptKeysCommand.java [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        Util.prohibitExternalCatalog(ctx.getDefaultCatalog(), this.getClass().getSimpleName());
+        DatabaseIf db = ctx.getCurrentCatalog().getDbOrAnalysisException(dbName);
+        List<List<String>> resultRowSet = Lists.newArrayList();
+        if (db instanceof Database) {
+            List<EncryptKey> encryptKeys = ((Database) db).getEncryptKeys();
+            List<List<Comparable>> rowSet = Lists.newArrayList();
+            for (EncryptKey encryptKey : encryptKeys) {
+                List<Comparable> row = encryptKey.getInfo();
+                // like predicate
+                if (wild == null || like(encryptKey.getEncryptKeyName().getKeyName())) {
```
Comment (LiBinfeng-01) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowEncryptKeysCommand.java:94]: it seems like we only support like in wildWhere predicate, so u can change syntax of DorisParser.g4 to support like only

File: fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisParser.g4 [hunk 4]
```diff
@@ -226,6 +226,7 @@ supportedShowStatement
         ((FROM | IN) database=identifier)?                                          #showView
     | SHOW PLUGINS                                                                  #showPlugins    
     | SHOW REPOSITORIES                                                             #showRepositories
+    | SHOW ENCRYPTKEYS ((FROM | IN) database=multipartIdentifier)? wildWhere?       #showEncryptKeys
```
Comment (LiBinfeng-01) [inline fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisParser.g4:229]:
```suggestion
    | SHOW ENCRYPTKEYS ((FROM | IN) database=multipartIdentifier)? (LIKE STRING_LITERAL)?       #showEncryptKeys    
```

Comment (msridhar78) [reply fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisParser.g4]: done - please review

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowEncryptKeysCommand.java (hunk 3 above)
Comment (msridhar78) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowEncryptKeysCommand.java]: done - pls review

**Comment by msridhar78** [comment]
run buildall

**Review by starocean999 — APPROVED** [review]

**Pull request merged**